src/conchat.js

```diff
@@ -142,14 +149,43 @@ class Con {
 
     await set(newUserRef, {
       username: this.#username,
-      room: this.#currentRoom,
+      room: this.#currentRoomKey,
     }).catch((error) => {
       console.error('Error adding user:', error);
     });
 
     this.#userKey = newUserRef.key;
   }
 
+  async #createNewRoom(roomName) {
+    const roomsRef = this.#getRef('chats/rooms');
+    const newRoomRef = push(roomsRef);
+
+    set(newRoomRef, {
+      name: roomName,
+      userList: [this.#username],
+    });
+
+    this.#currentRoomKey = newRoomRef.key;
+  }
+
+  async #isRoomValid(roomName, roomKey) {
+    const roomsRef = this.#getRef('chats/rooms');
+    const snapshot = await get(roomsRef);
+
+    let isValidKey = false;
+
+    snapshot.forEach((childSnapshot) => {
+      const roomData = childSnapshot.val();
+
+      if (roomData.name === roomName && childSnapshot.key === roomKey) {
+        isValidKey = true;
+      }
+    });
+
+    return isValidKey;
+  }
```

Comment:
전체 방 목록을 조회한 후 필요한 방의 정보를 찾는 방식의 로직인 것 같습니다.

방의 리스트가 많아진다면 데이터베이스에 부담을 줄 수 있어 성능을 저하시킬 수 있습니다! 

대신 Firebase의 [`DocumentSnapshot`](https://firebase.google.com/docs/reference/android/com/google/firebase/firestore/DocumentSnapshot)에 관련한 기능을 사용해서 필요한 방을 조회할 수 있을 것 같은데요. 
이 방법도 살펴봐주세요~

```suggestion
  async #isRoomValid(roomName, roomKey) {
  const roomRef = this.#getRef(`chats/rooms/${roomKey}`);
  const snapshot = await get(roomRef);

  return snapshot.exists() && snapshot.val().name === roomName;
}

```

Reply:
Firebase가 제공하는 메서드를 활용하여 데이터를 조회할 수 있는 방법을 잘 찾아주셨네요!
exists() 메서드를 활용하는 방법이 더욱 직관적이고 성능 측면에서도 효율적으로 보입니다. 리뷰주신 방법대로 수정하도록 하겠습니다~